src/util/group_pool.rs

```diff
@@ -0,0 +1,484 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::usize;
+use std::sync::{Arc, Mutex};
+use std::sync::mpsc::{Sender, Receiver};
+use std::sync::mpsc;
+use std::thread::Builder;
+use std::result::Result;
+use std::boxed::FnBox;
+use super::HashMap;
+
+
+type Thunk<'a> = Box<FnBox() + Send + 'a>;
```

Comment:
I copies the idea of `Thunk` from lib `ThreadPool`, it packages the `func` with it's arguments to run in the thread pool. @siddontang